tests/test_merge.py

```diff
@@ -51,33 +51,261 @@ def test_merge_branch(wst, capsys):
     assert out.split('\n')[0] == 'Merging 3.0.x into master'
 
 
-def test_merge_branch_with_whitelist(wst, capsys):
+def test_merge_branch_skip_last(wst, capsys):
     config.merge.branches = '1.0.x 2.0.x 3.0.x master'
+    with temp_git_repo():
+        # Dummy commit
+        run('git commit --allow-empty -m dummy-commit')
+
+        run('git checkout -b 3.0.x')
+
+        # commit 1
+        run('touch temp1.xml')
+        run('git add -A')
+        run('git commit -m commit1')
+
+        # commit 2
+        run('touch temp2.xml')
+        run('git add -A')
+        run('git commit -m  commit2')
+
+        # commit 3
+        run('touch temp3.xml')
+        run('git add -A')
+        run('git commit -m  [skip]')
+
+        run('git checkout master')
+
+        wst('merge 3.0.x --skip-commits \'skip\' \'space separated example\'')
 
+        changes = run('git log --oneline', return_output=True)
+        # Change should have been in the log entry
+        #assert 'Merge branch \'3.0.x\'' in changes
+        #assert '[skip]' in changes
+        #assert 'commit2' in changes
+
+        from os import listdir, popen
+        temp_dir = popen('pwd').read().strip()
```

Comment:
this is an interesting way to do this, but hoping we can do something a little simpler. Is it possible to just get the full absolute path of the temp_git_repo? Then we can construct the path to each file and use os.path.isfile to check if it exists.

Reply:
since temp_dir changes directories to the tmp folder it creates the current working directory is the tmp folder. so you could just look for the files relative to your current location, or you can just get the current working directory to construct the file paths.  temp_dir also created a variable with the tmp directory path that i think you have access to as well. lots of options here.